README.md

```diff
@@ -38,6 +40,86 @@ hop subscribe kafka://hostname:port/gcn -s EARLIEST
 This will read messages from the gcn topic from the earliest offset
 and read messages until an end of stream (EOS) is received.
 
+#### Python API
+
+Publish messages:
+
+Using the python API, we can publish various types of messages, including
+structured messages such as GCN Circulars and VOEvents:
+
+```python
+    from hop import stream
+    from hop.models import GCNCircular
+
+    # read in a GCN circular
+    with open("path/to/circular.gcn3", "r") as f:
+        circular = GCNCircular.load(f)
+
+    with stream.open("kafka://hostname:port/topic", "w") as s:
+        s.write(circular)
+```
+
+In addition, we can also publish unstructured messages as long as they are
+JSON serializable:
+
+```python
+    from hop import stream
+
+    with stream.open("kafka://hostname:port/topic", "w") as s:
+        s.write({"my": "message"})
+```
+
+By default, authentication is enabled for the Hop broker. In order to authenticate, one
+can pass in an `Auth` instance with credentials:
+
+```python
+    from hop import stream
+    from hop.auth import Auth
+
+    auth = Auth("my-username", "my-password")
+
+    with stream.open("kafka://hostname:port/topic", "w", auth=auth) as s:
+        s.write({"my": "message"})
+```
+
+A convenience function is also provided to read in auth configuration in the same way
+as in the CLI client:
+
+```python
+    from hop import stream
+    from hop.auth import load_auth
+
+    with stream.open("kafka://hostname:port/topic", "w", auth=load_auth()) as s:
+        s.write({"my": "message"})
+```
+
+Consume messages:
+
+
+```python
+    from hop import stream
+
+    with stream.open("kafka://hostname:port/topic", "r") as s:
+        for message in s:
+             print(message)
+```
+
+This will listen to the Hop broker, listening to new messages and printing them to
+stdout as they arrive until there are no more messages in the stream.
+By default, this will only process new messages since the connection was opened.
+The `start_at` option lets you control where in the stream you can start listening
+from. For example, if you'd like to listen to all messages stored in a topic, you can do:
+
+```python
+    from hop import stream
+    from hop.io import StartPosition
+
+    with stream.open("kafka://hostname:port/topic", "r", start_at=StartPosition.EARLIEST) as s:
+        for message in s:
+             print(message)
+```
+
+
 ## Installation
```

Comment:
I think we need the installation section first before getting into the details of authorization and APIs.

Reply:
I've been mostly following how some other projects (e.g. https://github.com/robinhood/faust) structure their sections. Not sure why they put their installation section later, but I'm happy to change it if you prefer it up top.

Reply:
I'd prefer it if we move it upward, it makes more sense to me. If you wish, we can get another opinion :)
I looked at https://github.com/fastavro/fastavro for instance or https://github.com/marcosschroh/python-schema-registry-client

Reply:
Nah that's fine, after rearranging the sections I think it looks better also. I updated the README accordingly.